### PR TITLE
feat: support Imagen model

### LINKED ISSUE #36: Support Imagen model
See [the documentation](https://console.cloud.google.com/vertex-ai/publishers/google/model-garden/imagegeneration)

### PR DESCRIPTION
**Pull request by adubovik**

### CODE REVIEW

**Comment by adubovik** [comment]
/deploy-review
>Deployment status: success

**Comment by adubovik** [comment]
/deploy-review
>Deployment status: success

**Comment by adubovik** [comment]
/deploy-review
>Deployment status: success

File: aidial_adapter_vertexai/llm/imagen_chat_completion_adapter.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    @staticmethod
+    def get_image_type(image: PIL_Image.Image) -> str:
+        match image.format:
+            case "JPEG":
+                return "image/jpeg"
+            case "PNG":
+                return "image/png"
+            case _:
+                raise ValueError(f"Unknown image mode: {image.mode}")
```
Comment (Oleksii-Klimov) [inline aidial_adapter_vertexai/llm/imagen_chat_completion_adapter.py]: Unknown image mode or format?

File: aidial_adapter_vertexai/universal_api/errors.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from typing import Mapping
+
+from aidial_sdk.chat_completion import Choice, Status
+from aiohttp import hdrs
+
+from aidial_adapter_vertexai.llm.exceptions import UserError
+
+
+def report_user_error(
```
Comment (Oleksii-Klimov) [inline aidial_adapter_vertexai/universal_api/errors.py]: Is this function used anywhere?

Comment (adubovik) [reply aidial_adapter_vertexai/universal_api/errors.py]: Removed along with a few other pieces of dead code.

File: aidial_adapter_vertexai/llm/imagen_chat_completion_adapter.py (hunk 1 above)
Comment (adubovik) [reply aidial_adapter_vertexai/llm/imagen_chat_completion_adapter.py]: Format indeed. Fixed.

**Review by Oleksii-Klimov — APPROVED** [review]

**Pull request merged**